hypothesis-python/tests/conjecture/test_engine.py

```diff
@@ -164,6 +164,7 @@ def recur(i, data):
         recur(i - 1, data)
 
 
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
 def test_recursion_error_is_not_flaky():
```

Comment:
The corollary is that recursion error *is* now flaky with the default pytest settings:
https://github.com/HypothesisWorks/hypothesis/actions/runs/9075635376/job/24936651302

We could bump the recursion limit in the callback, but that will probably not be a 100% solution since we still need ~2 stack frames for the bumping.

Or we could maybe do something with `unraisable_exception_runtest_hook` in the pytest plugin to selectively ignore it?

Reply:
Catching the `RecursionError` at top level callback seems to be sufficient for the tests to pass, but there may still be the risk of getting unlucky and violate the recursion limit at exactly top level.

*If* we accept this risk, we should maybe still put in the ignore-unraisable in our recursion self tests. Or wait and see if it does actually happen.